media-proxy/src/proxy_context.cc

```diff
@@ -944,15 +980,60 @@ int ProxyContext::RxStart(const mcm_conn_param* request)
     st_ctx->id = memif_ops.m_session_count;
     std::cout << "session id: " << st_ctx->id << std::endl;
     st_ctx->type = RX;
-    mStCtx.push_back(st_ctx);
+    mDpCtx.push_back(st_ctx);
 
     return (st_ctx->id);
 }
 
-int ProxyContext::TxStart(const mcm_conn_param* request)
+int ProxyContext::RxStart(const mcm_conn_param* request)
+{
+    if (request->payload_type == PAYLOAD_TYPE_RDMA_VIDEO) {
+        return RxStart_rdma(request);
+    } else {
+        return RxStart_mtl(request);
+    }
```

Comment:
```suggestion
    if (request->payload_type == PAYLOAD_TYPE_RDMA_VIDEO)
        return RxStart_rdma(request);

    return RxStart_mtl(request);
```

Reply:
I could argue that my version was little bit more readable, but I used your suggestion